activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -95,11 +95,27 @@ def self.[](*args)
     #   hash[:key] = 'value'
     #
     # This value can be later fetched using either +:key+ or <tt>'key'</tt>.
+    #
+    # If the value is a Hash or contains one or multiple Hashes, they will be
+    # converted to +HashWithIndifferentAccess+.
     def []=(key, value)
       regular_writer(convert_key(key), convert_value(value, conversion: :assignment))
     end
 
-    alias_method :store, :[]=
```

Comment:
I'm tempted to claim that the documentation might be clearer if we just add the kwarg to `[]=` and keep the alias, even though it'll only be usable (with standard syntax) via the alias.

(On the other hand, maybe we don't want to document the option at all for now...)

Reply:
I think we probably should offer that option. I'd rather not add it to `[]=` as I suspect it's a hotspot.

The first version of this PR exposed a new `raw_store` method rather than a keyword arg. Not sure which is better.

But either way, offering a way to skip that conversion make sense to me.